Test fix pull request

EKarpulevich opened the pull request:
Test system are working from eclipse now.
Also you can run all existing tests from test/Main.java

Commit:
Add tests dependency, fix err in test.

EKarpulevich committed:
ReilInstruction comparison fixed

EKarpulevich committed:
Fixed tests wrong behaviour.

Failed tests now don't put trash in db tables.

EKarpulevich committed:
Ignore strange test

EKarpulevich committed:
Test reuse fix

EKarpulevich committed:
Fixed jump addresses in cmpxchg

EKarpulevich committed:
Cast error fix

EKarpulevich committed:
prototype field from bn_functions added to tests

EKarpulevich committed:
junit 4 added to pom

EKarpulevich committed:
Move destroy tests to the end

EKarpulevich committed:
Add main method for all tests

EKarpulevich committed:
Info about test build added to readme

0xFFFFFFFE commented on src/test/java/com/google/security/zynamics/binnavi/testdata/database_schema.sql:
```diff
@@ -3450,41 +3450,57 @@ ALTER FUNCTION public.load_expression_type_instances(moduleid integer) OWNER TO
-    LANGUAGE sql
-    AS $_$
+CREATE OR REPLACE FUNCTION load_function_information(IN moduleid integer, IN address bigint)
+  RETURNS TABLE(
```
Can you go into a little bit of detail why this is necessary beneficial?

0xFFFFFFFE commented on src/test/java/com/google/security/zynamics/binnavi/testdata/database_schema.sql:
```diff
@@ -3497,80 +3513,81 @@ bc.bbcount, ec.edgecount, ic.incount, oc.outcount,
-$_$;
-
-
-ALTER FUNCTION public.load_function_information(moduleid integer) OWNER TO postgres;
```
Is there a reason to delete the ALTER to OWNER statements?

EKarpulevich committed:
Return missing ALTER TABLE

EKarpulevich commented:
@0xFFFFFFFE 
About this:

> > -CREATE FUNCTION load_function_information
> > +CREATE OR REPLACE FUNCTION load_function_information
> 
> Can you go into a little bit of detail why this is necessary beneficial?

There was errors in several tests because "bn_functions" table in tests had no "prototype" field and "load_function_information" function had no return field "prototype".
So I added one empty field to "bn_functions" table and take"load_function_information" function from  src/main/java/com/google/security/zynamics/binnavi/data/postgresql_tables.sql instead of wrong one.

And I just fixed missing ALTER OWNER statements.

EKarpulevich commented:
Currently sent for improvement.
